modules/http4s/test/src/smithy4s/http4s/Http4sPizzaSpec.scala

```diff
@@ -1,43 +0,0 @@
-/*
- *  Copyright 2021-2022 Disney Streaming
- *
- *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
- *  you may not use this file except in compliance with the License.
- *  You may obtain a copy of the License at
- *
- *     https://disneystreaming.github.io/TOST-1.0.txt
- *
- *  Unless required by applicable law or agreed to in writing, software
- *  distributed under the License is distributed on an "AS IS" BASIS,
- *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- *  See the License for the specific language governing permissions and
- *  limitations under the License.
- */
-
-package smithy4s.http4s
-
-import cats.effect.IO
-import cats.effect.Resource
-import org.http4s.Uri
-import org.http4s.client.Client
-import org.http4s.implicits._
-import smithy4s.example.PizzaAdminService
-
-object Http4sPizzaSpec extends smithy4s.tests.PizzaSpec {
```

Comment:
Unless I'm mistaken, I believe there might be tests in PizzaSpec that cannot be checked using compliance tests, that have to do with the `mapError` combinator. So I'd expect this suite to exist, but I'd expect a lot of tests to be eliminated from the `PizzaSpec` trait.

Reply:
Ok Then I will go through one by one and prune

Reply:
@Baccata I think most of the pizza spec will still be required , not just error transformation as the tests are testing specifically the http4s implementation not the protocol, i.e. a not found path returns a 404 or certain requests produce certain responses. Perhaps we should just leave it as is rather then micro pruning it

Reply:
> testing specifically the http4s implementation not the protocol

isn't that what all the protocol tests do? Test the interpreter for conforming to the protocol.

Reply:
no, specifically aws protocol tests , test Smithy-Model -> HTTP and vise versa , not HTTP -> HTTP

Reply:
Here are the tests I think can be removed in favour of smithy-defined tests : 

* https://github.com/disneystreaming/smithy4s/blob/series/0.17/modules/tests/src/smithy4s/tests/PizzaSpec.scala#L405-L433
* https://github.com/disneystreaming/smithy4s/blob/series/0.17/modules/tests/src/smithy4s/tests/PizzaSpec.scala#L70-L83
* https://github.com/disneystreaming/smithy4s/blob/series/0.17/modules/tests/src/smithy4s/tests/PizzaSpec.scala#L85-L117
* https://github.com/disneystreaming/smithy4s/blob/series/0.17/modules/tests/src/smithy4s/tests/PizzaSpec.scala#L141-L161
* https://github.com/disneystreaming/smithy4s/blob/series/0.17/modules/tests/src/smithy4s/tests/PizzaSpec.scala#L335-L357
* https://github.com/disneystreaming/smithy4s/blob/series/0.17/modules/tests/src/smithy4s/tests/PizzaSpec.scala#L384-L403
* https://github.com/disneystreaming/smithy4s/blob/series/0.17/modules/tests/src/smithy4s/tests/PizzaSpec.scala#L510-L543